packages/frontend/src/components/ChooseConnectionDropdown/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { IApp } from '@plumber/types'
+
+import { useCallback, useState } from 'react'
+import { FormControl } from '@chakra-ui/react'
+import { BxPlus, FormLabel, SingleSelect } from '@opengovsg/design-system-react'
+import AddAppConnection from 'components/AddAppConnection'
+
+type ConnectionDropdownOption = {
+  label: string
+  value: string
+  icon?: React.ElementType
+}
+interface ChooseConnectionDropdownProps {
+  isDisabled: boolean
+  connectionOptions: ConnectionDropdownOption[]
+  onChange: (value: string, shouldRefetch: boolean) => void
+  value: string | undefined
+  application: IApp
+}
+
+const ADD_CONNECTION_VALUE = 'ADD_CONNECTION'
+const ADD_NEW_CONNECTION_OPTION: ConnectionDropdownOption = {
+  label: 'Add new connection',
+  icon: BxPlus,
+  value: ADD_CONNECTION_VALUE,
+}
+
+function ChooseConnectionDropdown({
+  isDisabled,
+  connectionOptions,
+  onChange,
+  value,
+  application,
+}: ChooseConnectionDropdownProps) {
+  const [showAddConnectionDialog, setShowAddConnectionDialog] = useState(false)
+
+  const onSelectionChange = useCallback(
+    (value: string) => {
+      if (value === ADD_CONNECTION_VALUE) {
+        setShowAddConnectionDialog(true)
+        return
+      }
+      onChange(value, false)
+    },
+    [onChange, setShowAddConnectionDialog],
+  )
+
+  const handleAddConnectionClose = useCallback(
+    (response: any) => {
```

Comment:
nit: `response` is technically a `Record<string, any>`

Reply:
not much more type safety but 👌🏻